modules/home-environment.nix

```
@@ -206,6 +206,13 @@ in
       '';
     };
 
+    home.sessionPath = mkOption {
+      type = with types; listOf str;
+      description = "Extra directories to add to the $PATH";
+      example = [ ".git/safe/../../bin" "some/other/path" ];
```

These aren't relative paths, right? Or relative to what are they? I find the examples a bit hard to grasp. Wouldn't a typical application of this option have `~/.local/bin` or so?

---

They are, in fact, relative paths! They can be whatever, after all. I should maybe change this to just be what I am currently using it for:

```suggestion
      example = [ ".git/safe/../../bin" "$(xdg.configHome}/emacs/bin" ];
```

---

Relative to _what_? To the current source file? To `hm-session-vars.sh`? Can you add that to the documentation? I like the second example, but I wouldn't know what the first example does.

---

if a relative path is in your `$PATH`, it is relative to your current working directory.

let's say I am working on a node app. if my `PATH` is `node_modules/.bin`, then when I am `cd`'d into the repo's directory, `REPO/node_modules/.bin` is available without using `npx` or similar. Instead of calling `npx eslint` or whatever, you just call `eslint`.

These values are just plopped into the PATH as-is, so there isn't really much to document other than that.

Maybe the description line can read:

`Extra directories appended to $PATH. Added as-is.` Maybe my example just doesn't make sense for most people's use case?

---

```suggestion
      example = [
        ".git/safe/../../bin" # Relative paths are expanded in the current working directory
        "$(xdg.configHome}/emacs/bin" # Store paths are expanded to absolute paths
        "~/.local/bin" # Bash and other modern shells expand ~ to your home directory
      ];
```
Often a well-documented example suffices. How about something like this?

> if a relative path is in your $PATH, it is relative to your current working directory.

Oh, I didn't know!

---

Ah yeah _much_ better, thank you :bow:

and yeah, it is a good trick sometimes :grin: